src/lib/peterportal.types.ts

```diff
@@ -127,3 +127,9 @@ export interface SectionGQL {
     number: string;
     type: string;
 }
+
+export interface PrerequisiteJSON {
+    [key: string]: PrerequisiteJSONNode[];
+}
+
+export type PrerequisiteJSONNode = PrerequisiteJSON | string;
```

Comment:
```suggestion
export type PrerequisiteTree = Record<string, PrerequisiteNode[]>
export type PrerequisiteNode = Prerequisite | string;
```

I think something like this would look cleaner; interfaces can get annoying sometimes because they're more verbose.